cunumeric/eager.py

```diff
@@ -431,6 +456,57 @@ def diag(self, rhs, extract, k, stacklevel):
             self.array[:] = np.diag(rhs.array, k)
             self.runtime.profile_callsite(stacklevel + 1, False)
 
+    def diag_helper(
+        self,
+        rhs,
+        diag_size,
+        offset,
+        naxes,
+        extract,
```

Comment:
`extract` argument not handled

Reply:
The logic was hidden in the `naxes` argument (it is `no extract` if naxes<0), but I will add check for `extract` argument  as well

Reply:
In that case you don't need to pass both pieces of information, since one can be derived from the other?

Reply:
Thank you, I have added the check for `no extract`